### PR TITLE
Proposal to simplify route management

### PR DESCRIPTION
**Pull request by grtlr**

### CODE REVIEW

File: src/bin/inx-chronicle/api/mod.rs [hunk 1]
```diff
@@ -37,33 +37,59 @@ pub use self::{
 
 pub const DEFAULT_PAGE_SIZE: usize = 100;
 
+#[derive(Clone, Debug, Default)]
+pub struct RegisteredRoutes(Vec<String>);
+
+impl RegisteredRoutes {
+    pub fn register(&mut self, route: impl Into<String>) -> String {
```
Comment (Alexandcoats) [inline src/bin/inx-chronicle/api/mod.rs:44]: Maybe just accept a `&str` and return it to avoid the allocation weirdness.

**Comment by grtlr** [comment]
Woahh, where did that come from? :trollface: What do you think about adding `ApiState` and making our own `Router` non-generic?

**Comment by grtlr** [comment]
So here is the thing:

* We currently don't need a more sophisticated implementation (because we don't do JWT-based returns or different nestings, and neither does Hornet).
* No one actually uses this route for more than checking which high-level APIs are available.

So I think we should follow YAGNI here, we still have the code in the repo, if we ever want to come back to it.

**Comment by grtlr** [comment]
Closing for now due to inactivity.

**Pull request closed without merging**